FishyTransport/Tests/FishyActorTransportTests/GeneratorTest.swift

```diff
@@ -0,0 +1,213 @@
+      import ArgumentParser
+      import Logging
+      
+      import func Foundation.sleep
+      import struct Foundation.Data
+      import class Foundation.JSONDecoder
+      
+      extension Chatter: FishyActorTransport.MessageRecipient {
+        enum _Message: Sendable, Codable {
+          case join(room: ChatRoom)
+          case chatterJoined(room: ChatRoom, chatter: Chatter)
+          case chatRoomMessage(String, from: Chatter)
+        }
+        
+        nonisolated func _receiveAny<Encoder, Decoder>(
+          envelope: Envelope, encoder: Encoder, decoder: Decoder
+        ) async throws -> Encoder.Output
+          where Encoder: TopLevelEncoder, Decoder: TopLevelDecoder {
+          let message = try decoder.decode(_Message.self, from: envelope.message as! Decoder.Input) // TODO: this needs restructuring to avoid the cast, we need to know what types we work with
+          return try await self._receive(message: message, encoder: encoder)
+        }
+        
+        nonisolated func _receive<Encoder>(
+          message: _Message, encoder: Encoder
+        ) async throws -> Encoder.Output where Encoder: TopLevelEncoder {
+          do {
+            switch message {
+            case .join(let room):
+              try await self.join(room: room)
+              return try encoder.encode(Optional<String>.none)
+      
+            case .chatterJoined(let room, let chatter):
+              try await self.chatterJoined(room: room, chatter: chatter)
+              return try encoder.encode(Optional<String>.none)
+      
+            case .chatRoomMessage(let message, let chatter):
+              try await self.chatRoomMessage(message, from: chatter)
+              return try encoder.encode(Optional<String>.none)
+            }
+          } catch {
+            fatalError("Error handling not implemented; \(error)")
+          }
+        }
+        
+        @_dynamicReplacement(for: _remote_join(room:))
+        nonisolated func _fishy_join(room: ChatRoom) async throws  {
+          let message = Self._Message.join(room: room)
+          return try await requireFishyTransport.send(message, to: self.id, expecting: Void.self)
+        }
+        
+        @_dynamicReplacement(for: _remote_chatterJoined(room:chatter:))
+        nonisolated func _fishy_chatterJoined(room: ChatRoom, chatter: Chatter) async throws  {
+          let message = Self._Message.chatterJoined(room: room, chatter: chatter)
+          return try await requireFishyTransport.send(message, to: self.id, expecting: Void.self)
+        }
+        
+        @_dynamicReplacement(for: _remote_chatRoomMessage(_:from:))
+        nonisolated func _fishy_chatRoomMessage(_ message: String, from chatter: Chatter) async throws  {
+          let message = Self._Message.chatRoomMessage(message, from: chatter)
+          return try await requireFishyTransport.send(message, to: self.id, expecting: Void.self)
+        }
+      }
+      """#,
+    ],
+  ]
+}
```

Comment:
Thanks for the test, though the location of it is not really right; "Transport" things are the runtime, not the sourcegen. There should be a test target for the Generator, this  test can live in it then.

Thanks!